lib/src/clojure_lsp/handlers.clj

```diff
@@ -360,7 +360,9 @@
       :execute-command
       ;; TODO move components upper to a common place
       (when-let [{:keys [edit show-document-after-edit]} (refactor command arguments components)]
-        (producer/publish-workspace-edit producer edit)
+        (some-> (producer/publish-workspace-edit producer edit)
+                ;; wait for client to apply edit before showing doc/moving cursor
+                deref)
```

Comment:
Shouldn't we deref [here](https://github.com/clojure-lsp/clojure-lsp/pull/955/files#diff-1793b003b94c68bc4b15e7fa7c0fd503012b02fc891b2045ddd82240051d1a99L367) as well?

Reply:
I suppose so. We wouldn't want to tell the editor to navigate to a file before it had been created. Will change and check that the integration tests still work.